### ISSUE TITLE
[ISSUE] Issue with `databricks_external_location` resource CANNOT update/delete 

### ISSUE THREAD

**Issue by aar6ncai** [databricks/terraform-provider-databricks#2188]
Hi there,

When I make a change to the external location config it either failed to `update` or `delete` the existing resources.

```
##[error]Error: cannot update external location: External Location has dependent catalogs, schemas, managed or external tables/volumes; use force option to update url anyway
```

```
##[error]Error: cannot delete external location: External Location has dependent catalogs, schemas, managed or external tables/volumes; use force option to delete anyway
```
In addtion, there is no `forece_destory` option for external location -> https://github.com/databricks/terraform-provider-databricks/blob/master/docs/resources/external_location.md#argument-reference

instead, this `force_destory` option only exists in the `catalog` resource which associated with external location update/deletion -> https://github.com/databricks/terraform-provider-databricks/blob/master/docs/resources/catalog.md#argument-reference

### Configuration
```hcl
resource "databricks_external_location" "some" {
  name = "external"
  url  = format("abfss://%s@%s.dfs.core.windows.net/",
    azurerm_storage_account.ext_storage.name,
    azurerm_storage_container.ext_storage.name)
  credential_name = databricks_storage_credential.external.id
  depends_on      = [ databricks_metastore_assignment.this ]
}
```

### Expected Behavior
Update/Recreate external location resources successfully 

### Actual Behavior
```hcl
 # module.databricks_metastore.databricks_external_location.external_abc must be replaced
-/+ resource "databricks_external_location" "external_abc" {
      ~ id              = "external" -> (known after apply)
      ~ metastore_id    = "xxxxxxxxxxxxxx" -> (known after apply)
      ~ name            = "external" -> "external_abc" # forces replacement
      - skip_validation = false -> null
      ~ url             = "abfss://abcd" -> "abfss://abcde"
        # (2 unchanged attributes hidden)
    }
```

### Steps to Reproduce

Update external location resources and apply the change

### Terraform and provider versions
```
    databricks = {
      source  = "databricks/databricks"
      version = "1.9.2"
    }
```

### Debug Output
N/A

**Comment by mgyucht** [comment]
Note: a similar thing happens when changing storage credentials, though with a slightly different error message:
```
╷
│ Error: cannot update storage credential: Storage credential 'mi_credential' has 0 directly dependent external table(s) and 1 dependent external location(s); use force option to update anyway.
│ 
│   with module.telemetry_catalog.databricks_storage_credential.access_connector,
│   on ../../modules/databricks-azure-telemetry-catalog/main.tf line 17, in resource "databricks_storage_credential" "access_connector":
│   17: resource "databricks_storage_credential" "access_connector" {
│ 
```

**Comment by nkvuong** [comment]
Closed by #2546
